From bdf9f888eb513159d5ec0dd3b3e7328cfdcebdc7 Mon Sep 17 00:00:00 2001
From: wtgtybhertgeghgtwtg <wtgtybhertgeghgtwtg@gmail.com>
Date: Mon, 28 Nov 2016 00:04:11 -0700
Subject: [PATCH 16/34] Bump `color` dependency version. (#7)

---
 package.json | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/package.json b/package.json
index c3a9bb8..19ebf1e 100644
--- a/package.json
+++ b/package.json
@@ -21,7 +21,7 @@
     "index.js"
   ],
   "dependencies": {
-    "color": "^0.9.0",
+    "color": "^0.11.4",
     "postcss": "^5.0.4"
   },
   "devDependencies": {
-- 
2.33.1

