From aa9d148a8625de7cc0cb05ba662a662bfdf6d7bf Mon Sep 17 00:00:00 2001
From: Antonio Laguna <sombragriselros@gmail.com>
Date: Wed, 17 Nov 2021 21:37:38 +0100
Subject: [PATCH 13/18] Consistent formatting

---
 index.js | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/index.js b/index.js
index 5a65255..1f0d16a 100644
--- a/index.js
+++ b/index.js
@@ -1,5 +1,5 @@
-const {parse} = require("postcss-values-parser");
-const Punctuation = require("postcss-values-parser/lib/nodes/Punctuation");
+const { parse } = require('postcss-values-parser');
+const Punctuation = require('postcss-values-parser/lib/nodes/Punctuation');
 
 // whether the value has a lab() or lch() matcher
 const gradientRegExp = /(repeating-)?(conic|linear|radial)-gradient\([\W\w]*\)/i;
-- 
2.33.1

