From 2819d55da4385e46819e2feda4964a52709773d8 Mon Sep 17 00:00:00 2001
From: wtgtybhertgeghgtwtg <wtgtybhertgeghgtwtg@gmail.com>
Date: Tue, 31 Jan 2017 23:22:13 -0700
Subject: [PATCH 097/186] Bump `balanced-match`. (postcss/postcss-custom-properties#57)

---
 package.json | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/package.json b/package.json
index 6db90bc..425d40b 100644
--- a/package.json
+++ b/package.json
@@ -17,7 +17,7 @@
     "index.js"
   ],
   "dependencies": {
-    "balanced-match": "~0.1.0",
+    "balanced-match": "^0.4.2",
     "postcss": "^5.0.0"
   },
   "devDependencies": {
-- 
2.33.1

