From 9e44e385850ebd49fb62694099e3245384ff3b2e Mon Sep 17 00:00:00 2001
From: Max Belsky <public.belsky@gmail.com>
Date: Tue, 15 Jun 2021 05:02:07 +0300
Subject: [PATCH 175/186] Upgrade postcss-values-parser to v5 (postcss/postcss-custom-properties#248)

---
 package.json | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/package.json b/package.json
index a7fe218..fcd3356 100644
--- a/package.json
+++ b/package.json
@@ -29,7 +29,7 @@
     "node": ">=10.0.0"
   },
   "dependencies": {
-    "postcss-values-parser": "^4.0.0"
+    "postcss-values-parser": "^5.0.0"
   },
   "peerDependencies": {
     "postcss": "^8.1.0"
-- 
2.33.1

