From 2cb5ec018f677b767eeb02fedc38010cc38b5955 Mon Sep 17 00:00:00 2001
From: Bogdan Chadkin <trysound@yandex.ru>
Date: Wed, 12 Aug 2015 17:00:35 +0300
Subject: [PATCH 08/31] Upgrade to PostCSS 4.1.x

---
 .jscsrc      |  5 -----
 index.js     |  7 ++++---
 package.json | 14 +++++++-------
 3 files changed, 11 insertions(+), 15 deletions(-)

diff --git a/.jscsrc b/.jscsrc
index 29720b3..9afb9dd 100644
--- a/.jscsrc
+++ b/.jscsrc
@@ -121,10 +121,5 @@
   "requireCapitalizedConstructors": true,
   "safeContextKeyword": "that",
   "requireDotNotation": true,
-  "validateJSDoc": {
-    "checkParamNames": true,
-    "checkRedundantParams": true,
-    "requireParamTypes": true
-  },
   "requireSpaceAfterLineComment": true
 }
diff --git a/index.js b/index.js
index 750ae64..13507fd 100755
--- a/index.js
+++ b/index.js
@@ -1,8 +1,9 @@
 /**
  * Module dependencies.
  */
-var color = require("color")
+var postcss = require("postcss")
 var helpers = require("postcss-message-helpers")
+var color = require("color")
 
 /**
  * Constantes
@@ -13,7 +14,7 @@ var DECIMAL_PRECISION = 100000 // 5 decimals
 /**
  * PostCSS plugin to transform hexa alpha colors
  */
-module.exports = function plugin() {
+module.exports = postcss.plugin("postcss-color-hex-alpha", function() {
   return function(style) {
     style.eachDecl(function transformDecl(decl) {
       if (!decl.value || decl.value.indexOf("#") === -1) {
@@ -25,7 +26,7 @@ module.exports = function plugin() {
       }, decl.source)
     })
   }
-}
+})
 
 /**
  * transform RGBA hexadecimal notations (#RRGGBBAA or #RGBA) to rgba().
diff --git a/package.json b/package.json
index 3f2fcad..f78c9c0 100644
--- a/package.json
+++ b/package.json
@@ -5,7 +5,7 @@
   "keywords": [
     "css",
     "postcss",
-    "postcss-plugins",
+    "postcss-plugin",
     "color",
     "colour",
     "hexa",
@@ -23,14 +23,14 @@
     "index.js"
   ],
   "dependencies": {
-    "color": "^0.7.1",
-    "postcss-message-helpers": "^1.1.0"
+    "color": "^0.10.1",
+    "postcss": "^4.1.16",
+    "postcss-message-helpers": "^2.0.0"
   },
   "devDependencies": {
-    "jscs": "^1.6.2",
-    "jshint": "^2.5.6",
-    "postcss": "^4.0.2",
-    "tape": "^3.0.0"
+    "jscs": "^2.1.0",
+    "jshint": "^2.8.0",
+    "tape": "^4.0.3"
   },
   "scripts": {
     "lint": "jscs *.js **/*.js && jshint . --exclude-path .gitignore",
-- 
2.33.1

