From f693341a396991cb94beccadca05899fc4070461 Mon Sep 17 00:00:00 2001
From: Jonathan Neal <jonathantneal@hotmail.com>
Date: Mon, 29 Jul 2019 06:30:19 -0400
Subject: [PATCH 25/38] Fix bug with tests for errors

---
 src/index.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/index.js b/src/index.js
index 1db09c8..30dc1be 100644
--- a/src/index.js
+++ b/src/index.js
@@ -90,7 +90,7 @@ getOptions().then(
 				const areExpectedErrors = test.errors === Object(test.errors) && Object.keys(test.errors).every(
 					key => test.errors[key] instanceof RegExp
 						? test.errors[key].test(error[key])
-					: test.error[key] === error[key]
+					: test.errors[key] === error[key]
 				);
 
 				if (!areExpectedErrors) {
-- 
2.33.1

