From ef27dceb2e76399767608d1cc8e9cb2c6f68153c Mon Sep 17 00:00:00 2001
From: Jonathan Neal <jonathantneal@hotmail.com>
Date: Tue, 15 Sep 2015 19:51:59 -0400
Subject: [PATCH 015/108] Require & in selector

---
 index.js     | 2 +-
 test/test.js | 9 +++++++--
 2 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/index.js b/index.js
index e6ba6fd..e0182aa 100644
--- a/index.js
+++ b/index.js
@@ -67,7 +67,7 @@ function transpileRule(rule) {
 			var newRule = postcss.rule();
 
 			// if atrule is nest
-			if (child.name === name) {
+			if (child.name === name && child.params.indexOf('&') !== -1) {
 				transpileNestRule(rule, newRule, child);
 
 				transpileRule(newRule);
diff --git a/test/test.js b/test/test.js
index 020d60b..8d51ce8 100644
--- a/test/test.js
+++ b/test/test.js
@@ -18,18 +18,23 @@ describe('postcss-nested', function () {
 			  'z { color: red } a, b { color: white } a c, a d, b c, b d { color: blue }');
 	});
 
+	it('requires &', function () {
+		check('z { color: red } a, b { color: white; @nest c, d { color: blue } }',
+			  'z { color: red } a, b { color: white; @nest c, d { color: blue } }');
+	});
+
 	it('unwraps css deep nesting', function () {
 		check('z { color: red } a, b { color: white; @nest & c, & d { color: blue; @nest & e, & f { color: black } } }',
 			  'z { color: red } a, b { color: white } a c, a d, b c, b d { color: blue } a c e, a c f, a d e, a d f, b c e, b c f, b d e, b d f { color: black }');
 	});
 
 	it('unwraps mixed nesting', function () {
-		check('a { color: red; @media { color: white; @nest b { color: blue } } }',
+		check('a { color: red; @media { color: white; @nest & b { color: blue } } }',
 			  'a { color: red } @media { a { color: white } a b { color: blue } }');
 	});
 
 	it('unwraps specified at-rules', function () {
-		check('a { color: red; @unknown test { color: white } } b { color: white; @phone { color: blue } @media { color: black; @nest c { color: yellow } } }',
+		check('a { color: red; @unknown test { color: white } } b { color: white; @phone { color: blue } @media { color: black; @nest & c { color: yellow } } }',
 			  'a { color: red; @unknown test { color: white } } b { color: white } @phone { b { color: blue } } @media { b { color: black } b c { color: yellow } }',
 			  { bubble: ['phone'] });
 	});
-- 
2.33.1

