From 63e3cb360c3f9aa9d8b431c012c9a8bd76561cef Mon Sep 17 00:00:00 2001
From: Tommi Finnilae <baijeri@gmail.com>
Date: Mon, 12 Oct 2015 12:54:37 +0300
Subject: [PATCH 30/89] Update README.md

Add missing parenthesis
---
 README.md | 1 +
 1 file changed, 1 insertion(+)

diff --git a/README.md b/README.md
index f163009..29b6cd5 100644
--- a/README.md
+++ b/README.md
@@ -12,6 +12,7 @@ a, b {
 
     @nest c, d {
         color: white;
+    }
 
     @nest & & {
         color: blue;
-- 
2.33.1

