From bc1e73c7c82f09f754b61436a606ce5c92478c89 Mon Sep 17 00:00:00 2001
From: Tommi Finnilae <baijeri@gmail.com>
Date: Mon, 12 Oct 2015 13:02:50 +0300
Subject: [PATCH 31/95] Update README.md

fix example code
---
 README.md | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/README.md b/README.md
index 29b6cd5..e7a5ef0 100644
--- a/README.md
+++ b/README.md
@@ -10,7 +10,7 @@
 a, b {
     color: red;
 
-    @nest c, d {
+    @nest & c, & d {
         color: white;
     }
 
-- 
2.33.1

