From f53deb8c2bc402b8574b47c12686e8d122f704cb Mon Sep 17 00:00:00 2001
From: Jonathan Neal <jonathantneal@hotmail.com>
Date: Sat, 20 Feb 2016 23:01:31 -0500
Subject: [PATCH 42/89] Properly concatenate at-rules

---
 index.js | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/index.js b/index.js
index 1ae04c3..63299f5 100644
--- a/index.js
+++ b/index.js
@@ -1,4 +1,5 @@
 var postcss = require('postcss');
+var comma   = postcss.list.comma;
 
 module.exports = postcss.plugin('postcss-nesting', function (opts) {
 	var bubble = ['document', 'media', 'supports'];
@@ -50,7 +51,11 @@ module.exports = postcss.plugin('postcss-nesting', function (opts) {
 					var selector = rule.selector;
 
 					if (root.type === 'atrule' && root.name === target.name && root.parent) {
-						target.params = root.params + ' and ' + target.params;
+						target.params = comma(root.params).map(function (params1) {
+							return comma(target.params).map(function (params2) {
+								return params1 + ' and ' + params2;
+							}).join(', ');
+						}).join(', ');
 
 						rule = root;
 						root = root.parent;
-- 
2.33.1

