From c45f5713792b179fd8e738cf4f752f42a70b8445 Mon Sep 17 00:00:00 2001
From: Jonathan Neal <jonathantneal@hotmail.com>
Date: Sat, 20 Feb 2016 23:06:28 -0500
Subject: [PATCH 43/96] Use curly braces in conditionals

---
 index.js | 9 +++++++--
 test.js  | 4 +++-
 2 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/index.js b/index.js
index 63299f5..12b0b55 100644
--- a/index.js
+++ b/index.js
@@ -5,8 +5,13 @@ module.exports = postcss.plugin('postcss-nesting', function (opts) {
 	var bubble = ['document', 'media', 'supports'];
 	var name   = 'nest';
 
-	if (opts && opts.bubble) bubble = bubble.concat(opts.bubble);
-	if (opts && opts.prefix) name   = '-' + opts.prefix + '-' + name;
+	if (opts && opts.bubble) {
+		bubble = bubble.concat(opts.bubble);
+	}
+
+	if (opts && opts.prefix) {
+		name = '-' + opts.prefix + '-' + name;
+	}
 
 	return function (css) {
 		css.walk(function (target) {
diff --git a/test.js b/test.js
index b919681..7f57984 100644
--- a/test.js
+++ b/test.js
@@ -62,7 +62,9 @@ Object.keys(tests).forEach(function (name) {
 			plugin.process(inputCSS, options).then(function (result) {
 				var actualCSS = result.css;
 
-				if (debug) fs.writeFileSync(actualPath, actualCSS);
+				if (debug) {
+					fs.writeFileSync(actualPath, actualCSS);
+				}
 
 				t.equal(actualCSS, expectCSS, message);
 
-- 
2.33.1

