From 27c3311462e518cc372cd85c2e86314af22f68a3 Mon Sep 17 00:00:00 2001
From: Jonathan Neal <jonathantneal@hotmail.com>
Date: Sat, 9 Sep 2017 21:16:43 -0400
Subject: [PATCH 62/99] Update gitignore

---
 .gitignore | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/.gitignore b/.gitignore
index b8fc2cd..995098a 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,8 +1,10 @@
 node_modules
+package-lock.json
 .*
+!.appveyor.yml
+!.editorconfig
 !.gitignore
 !.tape.js
 !.travis.yml
 *.log*
 *.result.css
-package-lock.json
-- 
2.33.1

