From 5d1806da958bc53cd5371e1163bf0929c78cfcac Mon Sep 17 00:00:00 2001
From: Jonathan Neal <jonathantneal@hotmail.com>
Date: Wed, 3 Nov 2021 08:32:39 -0400
Subject: [PATCH 093/105] add npmrc

---
 .gitignore | 3 ++-
 .npmrc     | 2 ++
 2 files changed, 4 insertions(+), 1 deletion(-)
 create mode 100644 .npmrc

diff --git a/.gitignore b/.gitignore
index 27eda45..8049f8e 100644
--- a/.gitignore
+++ b/.gitignore
@@ -8,8 +8,9 @@ yarn.lock
 .*
 !.bin
 !.editorconfig
-!.gitignore
 !.github
+!.gitignore
+!.npmrc
 !.rollup.js
 !.tape.js
 !.travis.yml
diff --git a/.npmrc b/.npmrc
new file mode 100644
index 0000000..abb8cff
--- /dev/null
+++ b/.npmrc
@@ -0,0 +1,2 @@
+package-lock=false
+tag-version-prefix=""
\ No newline at end of file
-- 
2.33.1

