From 0237618f47eee76912af647fa6bcd837d6f243d8 Mon Sep 17 00:00:00 2001
From: Jonathan Neal <jonathantneal@hotmail.com>
Date: Mon, 21 May 2018 20:56:02 -0400
Subject: [PATCH 033/103] Rename autoprefixer option

---
 index.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/index.js b/index.js
index 7480e9d..c2d89d1 100644
--- a/index.js
+++ b/index.js
@@ -18,7 +18,7 @@ export default postcss.plugin('postcss-preset-env', opts => {
 			? 5
 		: parseInt(opts.stage) || 0
 	: 2;
-	const autoprefixerOptions = Object(opts).autoprefixerOptions
+	const autoprefixerOptions = Object(opts).autoprefixer;
 
 	const stagedAutoprefixer = autoprefixer(Object.assign({ browsers }, autoprefixerOptions));
 
-- 
2.33.1

