From 84c570001e7fc076eefe1b9a0cfcc8454d905c26 Mon Sep 17 00:00:00 2001
From: Daniel Ruf <daniel.ruf@ueberbit.de>
Date: Tue, 13 Nov 2018 12:45:09 +0100
Subject: [PATCH 66/84] ci: test Node.js 6, 8, 10 and 11

---
 .travis.yml | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/.travis.yml b/.travis.yml
index e1f11a5..7f1424c 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -3,6 +3,9 @@
 language: node_js
 
 node_js:
+  - node
+  - 10
+  - 8
   - 6
 
 install:
-- 
2.33.1

